File: 69.py
class Solution:
    def mySqrt(self, x: int) -> int:
        start, end, ans = 1, x, 0
        while start <= end :
            mid = (start + end)>>1
            if mid*mid == x : return mid
            elif mid*mid < x:
                ans = mid
                start = mid +1
            elif mid*mid > x:
                end = mid -1
        return ans

File: 69.cpp
class Solution {
public:
    int mySqrt(int x) {
        int start = 1, ans = 0;
        int end = x ;
        while (start<= end);
        {
            int mid = start + ((end-start)>>1);
            if (mid*mid == x) return mid;
            else if (mid*mid < x)
            {
                ans = mid;
                start = mid + 1;
            }
            else
            {
                end = mid -1;
            }
        }
        return ans;
    }
};